flag.go

```diff
@@ -542,7 +544,11 @@ func (f *FlagSet) parseArgs(args []string) (err error) {
 		}
 
 		if s[1] == '-' {
-			args, err = f.parseLongArg(s, args)
+			var terminate bool
+			args, terminate, err = f.parseLongArg(s, args)
+			if terminate {
```

Comment:
I'm not sure I love the idea of "terminate" and how intrusive this is...

You could maybe simply just check for `if len(s) == 2` here with a comment and get the same result, no?